packages/react-form-renderer/src/form-renderer/form-renderer.js

```diff
@@ -26,9 +26,20 @@ const FormRenderer = ({
   ...props
 }) => {
   const [fileInputs, setFileInputs] = useState([]);
+  const [registeredFields, setRegisteredFields] = useState({});
```

Comment:
What about to use ref to not trigger any renders?

Reply:
I don't think this triggers any additional renders of the children :thinking: Let me check the profiler. But, using ref might cause some race conditions because we can't use a callback to override the value.

Reply:
Oh yeah, it does :facepalm: we don't shield the render from parent re-renders. I'm just gonna slap the ref there and we will see what it does.

Reply:
OK it almost totally undid all the work. We can mimic the callback from setState which will make using ref safer.